assets/src/edit-story/app/font/fontProvider.js

```diff
@@ -88,6 +88,15 @@ function FontProvider({ children }) {
             name: fontWeightNames[weight],
             value: weight,
           }));
+
+          // It avoid reset Font Weight Dropdown when Bold button is enabled and Font Family don't have a Bold variant.
+          if (
+            fontWeights.filter((fontWeight) => {
+              return fontWeight.value === '700';
+            }).length === 0
+          ) {
```

Comment:
```suggestion
          const fontHasBold = fontWeights.some(({ value }) => value === '700');
          if (!fontHasBold) {
```

Reply:
But what if the font has extra bold (800) or black/heavy (900)? Shouldn't selecting bold then select one of those rather than use faux-bold?

Though I don't actually know if there are any Google fonts without Bold but with Extra Bold/Heavy.

 CC @pbakaus